### DISCUSSION TITLE
python3.7 version airflow 2.6.2 docker-compose google bigquery error [OSError: Could not find lib geos_c or load any of its variants ['libgeos_c.so.1', 'libgeos_c.so'].]

### DISCUSSION THREAD

**Discussion by limhaneul12** [apache/airflow#32182] [Q&A]
I'm getting the same error as the title I wrote, but I'm not sure if it's just me. 
I'm running it with docker-compose , so I can't do it locally. and docker-compose code in airflow public documentation 

page --> https://airflow.apache.org/docs/apache-airflow/2.6.2/docker-compose.yaml 

**OS**
MAC M1 PRO 

**screen shot**
<img width="1222" alt="image" src="https://github.com/apache/airflow/assets/52487610/69eea913-01e1-4d3c-bfab-c44fa8a567f7">

**Comment by potiuk** [comment]
ARM/M1 issue. It should work fine when you switch to 2.6.2 image - I think you are using an earlier airflow image. 

Libgeos was added here: https://github.com/apache/airflow/pull/31437

If you want to have  an ARM image with earlier version of Airflow that works with libgeos, you can also build your own custom image from 2.6.2 dockerfile and choose airflow version of your choice. See the second example:

https://airflow.apache.org/docs/docker-stack/build.html#building-from-pypi-packages

**Comment by potiuk** [comment]
Or maybe we might need to also add libgeos_dev for runtime. In which case we might need another fix for 2.6.3.

Can you please try to build, tag and use image based on Dockerfile similar to this:

```
FROM apache/airflow:2.6.2

USER root
RUN apt-get update && apt-get install -y libgeos-dev && apt-get autoremove -yqq --purge && apt-get clean
```

And report if it fixed your problem (look inside the docker-compose and you will find instrunctions on how to change docker-compose so that it builds your local dockerfile to build the image from local Dockerfile.

**Reply by andresitorusz** [reply]
This solution works on `apache/airflow:2.5.0-python3.10` as well. Thanks!

**Comment by potiuk** [comment]
@limhaneul12 -> if it fixes the problem, I will prepare a PR to fix it permanently from 2.6.3 onwards

**Reply by limhaneul12** [reply]
thank you !!! :) 

you`re fixed code dockerfile  injection in 
<img width="1240" alt="image" src="https://github.com/apache/airflow/assets/52487610/9e96e29d-2bbf-4798-808d-81010aaa7849">
 
not error clear log text :) 
<img width="1708" alt="image" src="https://github.com/apache/airflow/assets/52487610/d68c46e0-8265-4c8f-9e28-fac8f1e03c98">

If you can't load from load_dll among the shapelly/gose 100 lines, why don't you inject the related library from compose because the dependency injection isn't working or you can't find it?

**Reply by potiuk** [reply]
Right! thanks for confirming it. Fix is coming for 2.6.3.

**Reply by potiuk** [reply]
PR here: https://github.com/apache/airflow/pull/32200

**Reply by potiuk** [reply]
> If you can't load from load_dll among the shapelly/gose 100 lines, why don't you inject the related library from compose because the dependency injection isn't working or you can't find it?

It's not "me" . The bigquery library is trying to load it when being imported on ARM - we have no influence on that behaviour
